test/executables/test_components/package_with_over_react_match/pubspec.yaml

```diff
@@ -0,0 +1,9 @@
+name: package_with_match
+version: 0.0.0
+private: true
+environment:
+  sdk: '>=2.4.0 <3.0.0'
+dev_dependencies:
+  over_react: ">=1.0.0 <5.0.0"
```

Comment:
I think it'd be a good idea to have test cases for the version ranges we expect to see in the wild:
- `^4.0.0`
- `^4.1.0`
- `>=4.0.0 <6.0.0`
- `>=4.1.0 <6.0.0`
- `^5.0.0`
- `^5.1.0`

Not that the ones you have aren't worth keeping, but these are the most important to get right.